### PR TITLE
Integration of cullTimes for reinforcements with NameLayer2.5.43

### PR DESCRIPTION
**Pull request by Scuwr**
For integration with new version of NameLayer2.5.43. Added ability for reinforcements to cull when players in the group are not active.

CullTime is measured in days and the default is 60 days. Tested on localhost test server with one player.

### CODE REVIEW

**Comment by Scuwr** [comment]
Made reinforcement damage an exponential function @Maxopoly @ProgrammerDan

**Comment by Scuwr** [comment]
Also, it's been tested at each edge case and at the base case.

**Comment by ttk2** [comment]
So is it doing breaks to the reinforcement based on inactivity?

**Comment by Scuwr** [comment]
Yes, please check out this issues thread for more information:
https://github.com/Civcraft/NameLayer/issues/165

Also be sure to merge and load my NameLayer update before merge Citadel.
On Feb 14, 2016 10:08, "ttk2" <notifications@github.com> wrote:

So is it doing breaks to the reinforcement based on inactivity?

—
Reply to this email directly or view it on GitHub
<https://github.com/Civcraft/Citadel/pull/166#issuecomment-183928860>.

File: src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java [hunk 1]
```diff
@@ -70,6 +78,15 @@ public boolean isBypassable(Player p){
 		return gp.isAccessible(type, PermissionType.BLOCKS);
 	}
 	
+	public int getDamageMultiplier(){
+		Timestamp ts = gm.getTimestamp(g.getName());
```
Comment (ProgrammerDan) [inline src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java]: I'd recommend the following:

Don't store a reference to GM. Get a fresh one as here: https://github.com/Civcraft/Citadel/pull/166/files#diff-dbd016df79694880ff7b2e0bc5870217R157

On that same line note that the standard is now pass the Group object to the Manager call. Make the change here and in Namelayer; don't pass the Name back, just the group object and let Namelayer GM pick what data from the group object to use for lookups.

File: src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java [hunk 2]
```diff
@@ -32,6 +39,7 @@ public PlayerReinforcement(Location loc, int health,
 		this.g = g;
 		this.stack = stack;
 		gp = NameAPI.getGroupManager().getPermissionforGroup(g);
+		gm = NameAPI.getGroupManager();
```
Comment (ProgrammerDan) [inline src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java]: See note below but let's not store the GM reference.

File: src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java (hunk 1 above)
Comment (ProgrammerDan) [inline src/vg/civcraft/mc/citadel/reinforcement/PlayerReinforcement.java]: Actually disregard the second bit. Carry on with the string, I suppose. 

Maybe what I should say is I'd _prefer_ that it be standardized around the group object, since we're storing that in the reinforcement class already. But what you have is fine for now.

**Comment by ProgrammerDan** [comment]
Ok check my comments and rebase against master so that the merge is clean.

Overall this looks fine otherwise. Reminder to self that https://github.com/Civcraft/NameLayer/pull/166 needs to be merged _first_ to prevent build errors.

**Comment by Scuwr** [comment]
https://github.com/Civcraft/Citadel/pull/167
